validation-test/stdlib/Lazy.swift.gyb

```diff
@@ -1283,5 +1297,22 @@ do {
   }
 }
 
-runAllTests()
+% for Traversal in TRAVERSALS:
+%		TraversalCollection = collectionForTraversal(Traversal)
+tests.test("${TraversalCollection}/lazy/flatMap") {
+	let sample = (0..<100)
+	let expected = sample.filter { $0 % 3 != 0 }.map { String($0) }
+	let base = Minimal${TraversalCollection}(elements: sample).lazy
+
+	var calls = 0
+	let mapped = sample.lazy.flatMap { (x) -> String? in
+		calls += 1
+		return x % 3 != 0 ? String(x) : nil
+	}
+	expectEqual(0, calls, "unexpected eagerness in \(base.dynamicType).flatMap")
+	
+	checkSequence(expected, mapped, resiliencyChecks: .none)
+}
```

Comment:
I think the test should go into `test/1_stdlib/Inputs/flatMap.gyb`.  Please also reuse the `flatMapToOptionalTests` dataset from `stdlib/private/StdlibCollectionUnittest/CheckSequenceType.swift`.